components/script/script_thread.rs

```diff
@@ -181,6 +185,8 @@ impl InProgressLoad {
             is_visible: true,
             url: url,
             origin: origin,
+            navigation_start: 0,
+            navigation_start_precise: 0.,
```

Comment:
Ok, thank you for rephrasing the question. I added the constellation code trying to address your suggestion of setting the navigation start time as the time at which the thread that communicates with the network thread initiates the fetch. Setting navigation start during `InProgressLoad` construction does not vary the value significantly, but it indeed simplifies the code, so I'll change it as you suggests. Thanks for the feedback and apologies for the back and forth.